server/intelligence-service/app/mentor/models.py

```
@@ -0,0 +1,520 @@
+
+class UIMessage(BaseModel):
+    """Message model that matches the TypeScript interface."""
+
+    id: str = Field(..., description="A unique identifier for the message")
+    role: Literal["system", "user", "assistant"]
+    metadata: Optional[Dict[str, Any]] = None
+    parts: List[UIMessagePart]
+    
+    @field_validator('parts', mode='before')
+    @classmethod
+    def validate_parts(cls, v):
```

Pylint (3.3.7)

[refactor] Too many local variables (16/15)

(R0914)